src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,40 +1,88 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { getUser } from '../../api';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions as ActionCurrentTodo } from '../../features/currentTodo';
 
 export const TodoModal: React.FC = () => {
+  const [user, setUser] = useState<User>();
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    try {
+      if (currentTodo?.userId) {
+        getUser(currentTodo.userId)
+          .then(setUser);
+      }
+    } catch (error) {
+      throw new Error(`User: ${currentTodo?.userId} not found. Error: ${error}`);
+    }
+  }, [currentTodo]);
+
+  if (!currentTodo) {
+    return (
+      <p>No todo found</p>
+    );
+  }
```

Comment:
move  on the 41 line

Reply:
i cant

Reply:
all returns should be at the end of the component. 
but ok, let it be